src/watertap_contrib/reflo/solar_models/zero_order/flat_plate_physical.py

```diff
@@ -0,0 +1,426 @@
+                * b.cp_test
+                / b.collector_area
+                * log(1 - b.FR * b.UL * b.collector_area / (b.mdot_test * b.cp_test))
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="ratio of FRta_use to FRta_test, D&B Eq. 6.20.3",
+        )
+        def eq_r(b, t):
+            return b.r == (
+                (
+                    b.inlet_block[t].flow_mass_phase_comp["Liq", "H2O"]
+                    * b.number_collectors
+                    * b.inlet_block[t].cp_mass_phase["Liq"]
+                    / (b.collector_area * b.number_collectors)
+                    * (
+                        1
+                        - exp(
+                            -b.collector_area
+                            * b.number_collectors
+                            * b.Fprime_UL
+                            / (
+                                b.inlet_block[t].flow_mass_phase_comp["Liq", "H2O"]
+                                * b.number_collectors
+                                * b.inlet_block[t].cp_mass_phase["Liq"]
+                            )
+                        )
+                    )
+                )
+                / (b.FR * b.UL)
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="useful net heat gain, not accounting for pipe heat losses or a heat exchanger, D&B Eq. 6.8.1",
+        )
+        def eq_Q_useful(b, t):
+            return b.Q_useful[t] == (
+                b.collector_area
+                * b.number_collectors
+                * b.r
+                * (
+                    b.FR * b.ta * b.G_total * b.ta
+                    - b.FR * b.UL * (b.inlet_block[t].temperature - b.T_amb)
+                )
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Constraint to calculate the outlet temperature from FPC",
+        )
+        def eq_outlet_temp(b, t):
+            return b.outlet_block[t].temperature == (
+                b.inlet_block[t].temperature
+                + b.Q_useful[t]
+                / b.inlet_block[t].flow_mass_phase_comp["Liq", "H2O"]
+                / b.inlet_block[t].cp_mass_phase["Liq"]
+            )
+
+        @self.Constraint(doc="Pump power")
+        def eq_P_pump(b):
+            return b.electricity == (b.pump_power / b.pump_eff)
+
+        @self.Constraint(self.flowsheet().config.time, doc="Useful heat generated")
+        def eq_heat(b, t):
+            return b.heat == b.Q_useful[t]
+
+        @self.Constraint(doc="Heat generated annually")
+        def eq_heat_annual(b):
+            return b.heat_annual == b.heat * pyunits.convert(
+                1 * pyunits.year, to_units=pyunits.hour
+            )
+
+        # Need to check this
+        @self.Constraint(doc="Heat load of sysem")
+        def eq_heat_load(b):
+            return b.heat_load == b.collector_area_total * b.FR * b.ta * b.G_max * b.ta
```

Comment:
Can also consider the heat loss component as in SAM and in the surrogate model, since the coefficent is already constructed, then
`b.heat_load == b.collector_area_total * b.FR * ( b.ta * b.G_max * b.ta - b.UL * b.rated_delta_T)`
where `b.rated_delta_T = 30 K` is the default temp difference for calculating rated capacity in SAM.